Parameters section of the docs added + Consistency Edits

pop opened the pull request:
refs #36

pop committed:
Initial parameters section of the docs added.

refs #36

pop committed:
Unified to the :something syntax, other changes

Transitioned everything from the <something> syntax to the :something
syntax.
Added an example UUID to avoid any potential confusion.
Made each GET and POST request a sub-section (for easy linking)

matthewrsj commented:
This is a big improvement in readability and consistency. +1

pop committed:
Added explicit note about :variables

pop commented:
@osuosl/devs @Kennric Does anybody else have strong opinions about this?

MorganEPatch commented:
I like a lot of the syntax changes, +1 there, although the parameters section seems to also replicate effort from #38. We should decide which one we like better.

pop committed:
rebased and added to table of parameters

pop commented:
Currently there is overlap in the docs since @LyonesGamer wrote a similar section. I can try to consolidate them both into one section. Thoughts @osuosl/devs ?

To clarify, both blocks of content are in the PR (I rebased with develop after PR #38 was merged) they are just in different parts of the document; this may cause confusion for readers.

**Covered by both:**
- User, Project, Start, and End parameters behavior definitions.

**Covered by this PR:**
- Table of parameters.
- Formatting changes.
- Include Deleted parameter behavior.
- Include Revisions parameter behavior.

**Covered by PR #38:**
- Explicit definition of behavior when multiple parameters are put together (this is mentioned in this PR, but not as a general rule, just for `start` and `end`).

pop committed:
Organized parameters and changed heading styling

athai commented on source/draft_api.rst:
```diff
@@ -216,62 +233,135 @@ GET Endpoints
+~~~~~~~~~~~~~~~
+
+``/times?user=:username``
+    Fitlers results to a set of time submitted entries by a specified user.
```
Filters is misspelled.

Also, 'time submitted entries' is really weird wording.

pop committed:
Spelling. Words are the wrost.

athai commented on source/draft_api.rst:
```diff
@@ -216,62 +233,135 @@ GET Endpoints
+    These are objects which have the 'deleted_at' parameter set to an ISO date
+    (i.e., a non-null value).
+
 In addition to the path parameters to request a single object instead of a list,
```
This paragraph seems unnecessary. Single query parameters are already explained above and multiple parameters are explained below.

athai commented on source/draft_api.rst:
```diff
@@ -216,62 +233,135 @@ GET Endpoints
+?include_deleted=:bool
+~~~~~~~~~~~~~~~~~~~~~~
+
+Includes deleted entries in the returned results.
```
This line is bolded. Is that intentional?

https://github.com/osuosl/timesync/blob/elijah/parameters/source/draft_api.rst#include-deleted-bool

pop replied:
That is how restructured text renders things of the

```
word or phrase
    definition of word or phrase
```

So it's not 100% intentional, but that structure is intentional since I am elaborating on a statement with a definition.

athai commented on source/draft_api.rst:
```diff
@@ -216,62 +233,135 @@ GET Endpoints
+?activity=          :activityslug           /times
+?start=             :date (iso format)      /times
+?end=               :date (iso format)      /times
+?revisions=         :bool                   - /times
```
?revisions -> ?include_revisions

MorganEPatch commented on source/draft_api.rst:
```diff
@@ -216,62 +233,135 @@ GET Endpoints
+                                            - /activities/:slug
+                                            - /projects/
+                                            - /projects/:slug
+include_deleted     :bool                   - /times
```
include_deleted -> ?include_deleted (for consistency)

MorganEPatch commented on source/draft_api.rst:
```diff
@@ -615,25 +711,22 @@ Response body:
-      "created_at": "2014-04-16",
-      "updated_at": "2014-04-18",
-      "deleted_at": null,
+      "owner": "example-user", "created_at": "2014-04-16", "updated_at": "2014-04-18", "deleted_at": null,
```
Was this supposed to be all on the same line?

MorganEPatch replied:
Also note that in a different PR we got rid of the owner field, so that will need to be done across this PR.

athai commented on source/draft_api.rst:
```diff
@@ -216,62 +233,135 @@ GET Endpoints
-  ignored.
-* For more information about errors, check the
-  :ref:`draft_errors<draft_errors>` docs.
+    $ GET /times?activity=gwm&activity=pgd&token=...
```
Change activity -> project

pop committed:
Slight compatibitliy tweaks with develop

Also did a rebase with develop, probably missed something or goofed
the merge. Let me know if you find any inconsistencies.

pop commented:
@Kennric @osuosl/devs I rebased this with develop, lets get it merged asap.

athai commented on source/draft_api.rst:
```diff
@@ -214,62 +231,130 @@ GET Endpoints
+
+.. code-block:: none
+
+    $ GET /times?activity=gwm&project=pgd&token=...
```
Change activity to project; gwm isn't an activity
